cibuildwheel/macos.py

```diff
@@ -50,7 +51,14 @@ def call(args, env=None, cwd=None, shell=False):
         return subprocess.check_call(args, env=env, cwd=cwd, shell=shell)
 
     # get latest pip once and for all
-    call(['curl', '-L', '-o', get_pip_script, get_pip_url])
```

Comment:
`curl` has a retry option. This would be simpler to use it I guess:

```
Matt$ curl --help | grep retry
     --retry NUM   Retry request NUM times if transient problems occur
     --retry-connrefused  Retry on connection refused (use with --retry)
     --retry-delay SECONDS  Wait SECONDS between retries
     --retry-max-time SECONDS  Retry only within this period
```